v2/devdoc/log_context_property_basic_types_requirements.md

```diff
@@ -29,392 +29,585 @@ Note: More types will be added as needed.
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_002: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` shall call `snprintf` with `buffer`, `buffer_length` and format string `PRId64` and pass in the values list the `int64_t` value pointed to be `property_value`. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_003: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string` shall succeed and return the result of `snprintf`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_003: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` shall succeed and return the result of `snprintf`. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_004: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).to_string` shall fail and return a negative value. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_004: [** If any error is encountered, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).to_string` shall fail and return a negative value. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy
 
 ```c
 typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* dst_value, void* src_value);
 ```
 
-`LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` copies the `int64_t` value from the address pointed by `src_value` to `dst_value`.
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` copies the `int64_t` value from the address pointed by `src_value` to `dst_value`.
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_005: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_005: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_006: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_006: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall fail and return a non-zero value. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_007: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall copy the bytes of the `int64_t` value from the address pointed by `src_value` to `dst_value`. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_007: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall copy the bytes of the `int64_t` value from the address pointed by `src_value` to `dst_value`. **]**
 
-**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).copy` shall succeed and return 0. **]**
+**SRS_LOG_CONTEXT_PROPERTY_BASIC_TYPES_01_008: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).copy` shall succeed and return 0. **]**
 
-## LOG_CONTEXT_PROPERTY_TYPE_IMPL(int64_t).free
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(int64_t).free
 
 ```c
 typedef int (*LOG_CONTEXT_PROPERTY_TYPE_FREE)(void* value);
```

Comment:
>int [](http://example.com/codeflow?start=8&length=3)

Interesting that a `free` function wants to return "something" (in this case, some int).

Also 01_009 should specify the exact value that the function is returning. "shall return... what?"

Reply:
or is the "int" in line 74 actually "void"?

Reply:
It is supposed to be void. It is void in the code, for some reasons the .mds were out of sync